shared-kernel-infrastructure/src/main/java/org/dcsa/skernel/infrastructure/http/headers/api/version/DcsaSpecificationConfiguration.java

```diff
@@ -0,0 +1,85 @@
+package org.dcsa.skernel.infrastructure.http.headers.api.version;
+
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.*;
+import java.util.regex.Pattern;
+
+@Configuration
+@ConfigurationProperties("dcsa.specification")
+@Setter
+@Getter
+public class DcsaSpecificationConfiguration {
+
+  static final String UNOFFICIAL = "THIS-CONTEXT-PATH-IS-AN-UNOFFICIAL-API";
+  static final Pattern CONTEXT_PREFIX_PATTERN = Pattern.compile(
+    // Match with up to 5 leading contexts (to avoid a run-away regex).
+    "^((?:/+[^/\\s]+){0,5}/+v\\d+)(?:/.*)?$"
+  );
+
+  private String version;
+  // We use a list here as Spring "eats" the "/" in property keys.
+  private List<PrefixSpecificationRule> byPrefix;
+
+  private Map<String, String> byPrefixCache = Collections.emptyMap();
+
+  public void setByPrefix(List<PrefixSpecificationRule> byPrefix) {
```

Comment:
it is unclear to me when setByPrefix is called or need to be called. And subsequently when the cache is (re)populated

Reply:
This is called by Spring as a part of the configuration provisioning (via `@ConfigurationProperties`) - it is using setters rather than the constructor for ensuring the correct state.